libs/queue/README.md

```diff
@@ -0,0 +1,64 @@
+# Overview
+
+The Queue library provides an implementation of a queue data structure. A Queue is a linear structure which follows the First-In-First-Out (FIFO) principle. This means that the elements added first are the ones that get removed first.
+
+The `Queue<T>` is a structure with a generic type parameter T. The elements in the queue are stored as a `Vec<T>`. The data structure is created empty and is growable.
```

Comment:
```suggestion
The `Queue<T>` is a structure with a generic type parameter `T`. The elements in the queue are stored as a `Vec<T>`. The data structure is created empty and is growable.
```

The highlight is for consistency.

In terms of the content I would entirely remove the sentence or in-cooperate the use of the generic into some other place.
The reason is that as a user of the library you care about its utility rather than its implementation.
That internal representation would typically be encapsulated from the user similar to how an API informs you of how to use it rather than what it does under the hood (this also allows you to re-implement it without having to update docs here).

On the other hand, from the perspective of a developer you'd want to inform the reader of this kind of information. The place for that would be the specification rather than the more general "introductory" document which is the readme

Reply:
Good suggestion. Will make some adjustments.